lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -130,6 +140,82 @@ class ReportUIFeatures {
     dropdown.addEventListener('click', this.onExport);
   }
 
+  _setupThirdPartyFilter() {
+    const thirdPartyFilterAuditExclusions = ['uses-rel-preconnect'];
+
+    // get all tables with a text url
+    /** @type {Array<HTMLTableElement>} */
+    const tables = Array.from(this._document.querySelectorAll('.lh-table'));
+    const tablesWithUrls = tables
+      .filter(el => el.querySelector('td.lh-table-column--url'))
+      .filter(el =>!thirdPartyFilterAuditExclusions.includes(
+        /** @type {Element} */(el.closest('.lh-audit')).id));
+
+    tablesWithUrls.forEach((tableEl, index) => {
+      const thirdPartyRows = this._getThirdPartyRows(tableEl, this.json.finalUrl);
+      // no 3rd parties, no checkbox!
+      if (!thirdPartyRows.size) {
+        return;
+      }
+      // create input box
+      const filterTemplate = this._dom.cloneTemplate('#tmpl-lh-3p-filter', this._document);
+      const filterInput = /** @type {HTMLInputElement} */ (filterTemplate.querySelector('input'));
+      const id = `lh-3p-filter-label--${index}`;
+
+      filterInput.setAttribute('id', id);
+      filterInput.addEventListener('change', e => {
+        // remove elements from the dom and keep track of them to readd on uncheck
+        // why removing instead of hiding? nth-child(even) background-colors keep working
+        if (e.target instanceof HTMLInputElement && e.target.checked) {
+          for (const [position, row] of thirdPartyRows.entries()) {
+            const childrenArr = getTableRows(tableEl);
+            tableEl.tBodies[0].insertBefore(row, childrenArr[position]);
+          }
+        } else {
+          for (const position of thirdPartyRows.keys()) {
+            const row = /** @type {HTMLTableRowElement} */ (thirdPartyRows.get(position));
+            row.remove();
+          }
+        }
+      });
+
+      if (tableEl.parentNode) {
+        /** @type {Element} */ (filterTemplate.querySelector('label')).setAttribute('for', id);
+        /** @type {Element} */ (filterTemplate.querySelector('.lh-3p-filter-count')).textContent =
+          `${thirdPartyRows.size}`;
+        // Finally, add checkbox to the DOM
+        tableEl.parentNode.insertBefore(filterTemplate, tableEl);
+      }
+    });
+  }
+
+  /**
+   * @param {HTMLTableElement} el
+   * @param {string} finalUrl
+   * @return {Map<number, HTMLTableRowElement>}
+   */
+  _getThirdPartyRows(el, finalUrl) {
+    /** @type {NodeListOf<HTMLElement>} */
+    const urlItems = el.querySelectorAll('.lh-text__url');
+    const rootDomain = Util.getRootDomain(finalUrl);
+    /** @type {Map<number, HTMLTableRowElement>} */
+    const thirdPartyRows = new Map();
+    for (const urlItem of urlItems) {
+      const isThirdParty = !urlItem.title.includes(`${rootDomain}/`);
```

Comment:
heh, this feels a bit like a hack :) should we make a note to have a proper data attribute or something with the URL in case we want the title to be something else someday?

Reply:
I actually added it to the dataset right away.